postscripts/puppet_config_cached_catalog

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+
+###
+# Description: Configure puppet agent use_cached_catalog setting.
+#              Expects that puppet agent is already installed and configured.
+#              See also: postscripts/puppet_configure (in this repo)
+# Usage:
+#      1. updatenode <noderange> -P "puppet_config_cached_catalog [OPTS]"
+#
+# Source: https://github.com/ncsa/xcat-tools
+###
+
+
+# PREP
+PRG=$( basename $0 )
+logger -t xcat -p local4.info "running '$PRG' on node $NODE"
+
+
+# DEFAULT SETTINGS
+declare -A DEFAULTS
+DEFAULTS[ENABLE]=true
+
+
+# GLOBAL SETTINGS
+PUPPET=/opt/puppetlabs/bin/puppet
+AGENTLOG=/var/log/xcat/puppet_agent.log
+
+
+# FUNCTIONS
+logr() {
+  logger -t xcat -p local4.info "$*"
+  echo "$*"
+}
+
+
+croak() {
+  logr "ERROR - $*"
+  echo "ERROR - $*"
+  exit 99
+}
+
+
+enable_puppet_agent_use_cached_catalog() {
+    logr "Enabling puppet agent use_cached_catalog"
+    $PUPPET config set use_cached_catalog true --section agent
+    logr "Enabling puppet agent use_cached_catalog ... OK"
+}
+
+
+disable_puppet_agent_use_cached_catalog() {
+    logr "Disabling puppet agent use_cached_catalog"
+    $PUPPET config set use_cached_catalog false --section agent
+    logr "Disabling puppet agent use_cached_catalog ... OK"
+}
+
+
+print_usage() {
+    cat <<ENDHERE
+Usage:
+    puppet_config_cached_catalog [OPTIONS]
+
+OPTIONS:
+    -h | --help
+        Print usage message and exit
+    --true
+        Enable use_cached_catalog
+    --false
+        Disable use_cached_catalog
+
+ENDHERE
+}
+
+
+# DO WORK
+ENDWHILE=0
+while [[ $# -gt 0 ]] && [[ $ENDWHILE -eq 0 ]] ; do
+  case $1 in
+    -h|--help)
+        print_usage
+        exit 0
+        ;;
+    --true)
+        enable_puppet_agent_use_cached_catalog
+        ;;
+    --false)
+        disable_puppet_agent_use_cached_catalog
+        ;;
+    --)
+        ENDWHILE=1
+        ;;
+    -*)
+        croak "Invalid option '$1'"
+        ;;
+     *)
+        ENDWHILE=1
+        break
+        ;;
+  esac
+  shift
+done
```

Comment:
This while loop might not be the best approach if we're only expecting a single option variable.

As written now, I think it would support running as follows:
```
./puppet_config_cached_catalog --true --false --true --false
```
...and getting the following output:
```
Enabling puppet agent use_cached_catalog
Enabling puppet agent use_cached_catalog ... OK
Disabling puppet agent use_cached_catalog
Disabling puppet agent use_cached_catalog ... OK
Enabling puppet agent use_cached_catalog
Enabling puppet agent use_cached_catalog ... OK
Disabling puppet agent use_cached_catalog
Disabling puppet agent use_cached_catalog ... OK
...
```
That is probably confusing and unexpected.

Perhaps do one of the following:
- convert to only accept a single parameter
- or add an `exit 0` after the true/false cases